components/latex_block/latex_block.tsx

```diff
@@ -2,16 +2,20 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
+
 import {FormattedMessage} from 'react-intl';
 
-export default class LatexBlock extends React.PureComponent {
-    static propTypes = {
-        content: PropTypes.string.isRequired,
-        enableLatex: PropTypes.bool.isRequired,
-    }
+type Props = {
+    content: string;
+    enableLatex: boolean;
+};
 
-    constructor(props) {
+type State = {
+    katex: any;
```

Comment:
What should the type of katex be?

Reply:
Looks like `katex` doesn't provide its own types, but DefinitelyTyped does.
See here: https://www.npmjs.com/package/@types/katex

Reply:
Looks like you missed adding the type here.

Reply:
From https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/katex/index.d.ts, it looks like `katex` is being exported as a namespace. What type should I label `katex` as?

Reply:
Hmm, it's surprising that they don't export a type for the entire `katex` object, however they do export one for `renderToString`.

I propose we refactor the component a little bit to store `renderToString` in state instead of the whole `katex` object and we should be able to type that pretty easily.